saleor/payment/gateways/np_atobarai/api_helpers.py

```diff
@@ -0,0 +1,258 @@
+    path: str = "",
+    json: Optional[dict] = None,
+) -> requests.Response:
+    with np_atobarai_opentracing_trace("np-atobarai.utilities.request"):
+        response = requests.request(
+            method=method,
+            url=get_url(config, path),
+            timeout=REQUEST_TIMEOUT,
+            json=json or {},
+            auth=HTTPBasicAuth(config.merchant_code, config.sp_code),
+            headers={"X-NP-Terminal-Id": config.terminal_id},
+        )
+        if 400 < response.status_code <= 600:
+            raise requests.HTTPError
+        return response
+
+
+def np_request(
+    config: "ApiConfig", method: str, path: str = "", json: Optional[dict] = None
+) -> NPResponse:
+    try:
+        response = _request(config, method, path, json)
+        response_data = response.json()
+        if "errors" in response_data:
+            return NPResponse({}, response_data["errors"][0]["codes"])
+        return NPResponse(response_data["results"][0], [])
+    except requests.RequestException:
+        logger.warning("Cannot connect to NP Atobarai.", exc_info=True)
+        return NPResponse({}, [NP_CONNECTION_ERROR])
+
+
+def handle_unrecoverable_state(
+    order: Optional[Order],
+    action: str,
+    transaction_id: str,
+    error_codes: Iterable[str],
+) -> None:
+    message = f"Payment #{transaction_id} {action.capitalize()} Unrecoverable Error"
+    logger.error("%s: %s", message, ", ".join(error_codes))
+    if order:
+        notify_dashboard(order, message)
+
+
+def health_check(config: "ApiConfig") -> bool:
+    try:
+        _request(config, "post", "/authorizations/find")
+        return True
+    except requests.RequestException:
+        return False
+
+
+def format_name(ad: AddressData) -> str:
+    """Follow the Japanese name guidelines."""
+    return f"{ad.last_name}　{ad.first_name}".strip()
+
+
+def format_address(config: "ApiConfig", ad: AddressData) -> Optional[str]:
+    """Follow the Japanese address guidelines."""
+    # example: "東京都千代田区麹町４－２－６　住友不動産麹町ファーストビル５階"
+    if not config.fill_missing_address:
+        return f"{ad.country_area}{ad.street_address_1}{ad.street_address_2}"
+    with Posuto() as pp:
+        try:
+            jap_ad = pp.get(ad.postal_code)
+        except KeyError:
+            logger.warning(f"Invalid japanese postal code: {ad.postal_code}")
+            return None
+        else:
+            return (
+                f"{ad.country_area}"
+                f"{jap_ad.city}"
+                f"{jap_ad.neighborhood}"
+                f"{ad.street_address_1}"
+                f"{ad.street_address_2}"
+            )
+
+
+def format_price(price: Decimal, currency: str) -> int:
+    return int(price_to_minor_unit(price, currency))
+
+
+def get_refunded_goods(
+    config: "ApiConfig",
+    refund_data: Dict[int, int],
+    payment_information: PaymentData,
+) -> List[dict]:
+    return [
+        {
+            "goods_name": line.product_sku if config.sku_as_name else line.product_name,
```

Comment:
Will this work when `product_sku` is not known? SKUs are optional in 3.1.

Reply:
`sku_as_name` is a special (it's off by default) option for users that specifically use SKUs and want them in NP transactions.

Reply:
I am assuming that it's going to fail when we pass `null` - something has to be printed on an invoice. On the other hand, the requirement from the customer was that SKU is displayed as the invoice line description, therefore it was added as a plugin configuration option.

We need to make a judgement call here, should we leave it up to customer to take responsibility for changing the plugin configuration to use SKUs or fall back to the product name if SKU is unset? I'd probably vote for the latter.

Reply:
In similar cases, we assumed that we will use .id when .sku is missing. Currently, I think that we need to be sure that this will not raise an exception that we will not handle.